utils/src/store.rs

```diff
@@ -0,0 +1,21 @@
+use cid::Cid;
+use libipld::{
+  cbor::DagCborCodec,
+  codec::Codec,
+  ipld::Ipld,
+};
+use multihash::{
+  Code,
+  MultihashDigest,
+};
+
+/// This trait describes the interations with
+/// externaly stored IPLD structures. 
+pub trait Store: std::fmt::Debug {
+    fn put(&self, expr: Ipld) -> Cid;
+    fn get(&self, link: Cid) -> Option<Ipld>;
+}
+
+pub fn cid(x: &Ipld) -> Cid {
```

Comment:
I want to have a single source of truth for this hash function from `yatima_core::cid::cid` so that we don't accidentally introduce hashing errors